src/org/sipdroid/sipua/SipdroidEngine.java

```diff
@@ -131,295 +137,313 @@ UserAgentProfile getUserAgentProfile(SIPIdentity sipIdentity) {
+			} catch (Exception E) {
+				Log.e(getClass().getCanonicalName(), "Exception while setting up profile: " +
+						E.getClass().getCanonicalName() + ": " +
+						E.getMessage());
 			}
-			register();
-			listen();
+			i++;
+		}
+		register();
+		listen();
 
-			return true;
+		return true;
 	}
-	
+
 	private String getContactAddress() {
 		String addr = IpAddress.localIpAddress;
-		if(addr.indexOf(':') >= 0) {
+		if (addr.indexOf(':') >= 0) {
 			int zoneIndex = addr.indexOf('%');
-			if(zoneIndex < 0)
+			if (zoneIndex < 0)
 				zoneIndex = addr.length();
 			addr = '[' + addr.substring(0, zoneIndex) + ']';
 		}
-		
+
 		return addr;
 	}
 
-	private String getContactURL(String username,SipProvider sip_provider) {
+	private String getContactURL(String username, SipProvider sip_provider) {
 		int i = username.indexOf("@");
 		if (i != -1) {
 			// if the username already contains a @ 
 			//strip it and everthing following it
 			username = username.substring(0, i);
 		}
-		
-		if(username.length() == 0) {
+
+		if (username.length() == 0) {
 			username = "unspecified";
 		}
 
 		return username + "@" + getContactAddress()
-		+ (sip_provider.getPort() != 0?":"+sip_provider.getPort():"")
-		+ ";transport=" + sip_provider.getDefaultTransport();		
+				+ (sip_provider.getPort() != 0 ? ":" + sip_provider.getPort() : "")
+				+ ";transport=" + sip_provider.getDefaultTransport();
 	}
-	
-	void setOutboundProxy(SipProvider sip_provider,int i) {
+
+	void setOutboundProxy(SipProvider sip_provider, int i) {
 		try {
-			if (sip_provider != null) sip_provider.setOutboundProxy(new SocketAddress(
-					user_profiles[i].outbound_server_name,
-					user_profiles[i].outbound_server_port
-					));
+			if (sip_provider != null)
+				sip_provider.setOutboundProxy(new SocketAddress(
+						user_profiles[i].outbound_server_name,
+						user_profiles[i].outbound_server_port
+				));
 		} catch (Exception e) {
 		}
 	}
-	
+
 	public void CheckEngine() {
 		int i = 0;
 		for (SipProvider sip_provider : sip_providers) {
 			if (sip_provider != null && !sip_provider.hasOutboundProxy())
-				setOutboundProxy(sip_provider,i);
+				setOutboundProxy(sip_provider, i);
 			i++;
 		}
 	}
 
 	public Context getUIContext() {
 		return Receiver.mContext;
 	}
-	
+
 	public int getRemoteVideo() {
 		return ua.remote_video_port;
 	}
-	
+
 	public int getLocalVideo() {
 		return ua.local_video_port;
 	}
-	
+
 	public String getRemoteAddr() {
 		return ua.remote_media_address;
 	}
-	
+
 	public void expire() {
 		Receiver.expire_time = 0;
 		int i = 0;
 		for (RegisterAgent ra : ras) {
 			if (ra != null && ra.CurrentState == RegisterAgent.REGISTERED) {
 				ra.CurrentState = RegisterAgent.UNREGISTERED;
-				Receiver.onText(Receiver.REGISTER_NOTIFICATION+i, null, 0, 0);
+				Receiver.onText(Receiver.REGISTER_NOTIFICATION + i, null, 0, 0);
 			}
 			i++;
 		}
 		register();
+		for (PublishAgent pa : pas) {
+			if (pa != null) {
+				pa.publish();

```

Comment:
i don't think we want to publish at this stage,  this function seems to be call when the registration expires, in which case you want to un-publish, probably.